### PR TITLE
encode params before callConfigServer

### LINKED ISSUE #119: ConfigParam.Content 字符串中带有 & 无法推送到 nacos
在 PublishConfig 的时候，字符串中带有 & 的怎么处理呢？& 字符串后面的内容全部被丢弃了

### PR DESCRIPTION
**Pull request by mark4z**
Fixes #119

关于为什么没有Decode的部分：

Nacos server configController会解码params， 所以拉取配置时不必解码

### CODE REVIEW

File: util/http.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import "net/url"
+
+func EncodingParams(params map[string]string) map[string]string {
```
Comment (sanxun0325) [inline util/http.go:21]: maybe it's better way.
```
 func EncodingParams(params map[string]string) {
	for param, value := range params {
		params [param] = url.QueryEscape(value)
 	}
 }
```

Comment (sanxun0325) [reply util/http.go:21]: EncodingParams func should be used in http_agent#Request func

Comment (mark4z) [reply util/http.go]: You are right, it's a mistake,  maybe java has deeply affected me. 
And now, it run  expected

**Review by sanxun0325 — APPROVED** [review]

File: common/http_agent/http_agent.go [hunk 2]
```diff
@@ -71,6 +71,8 @@ func (agent *HttpAgent) RequestOnlyResult(method string, path string, header htt
 }
 
 func (agent *HttpAgent) Request(method string, path string, header http.Header, timeoutMs uint64, params map[string]string) (response *http.Response, err error) {
+	util.EncodingParams(params)
```
Comment (sanxun0325) [inline common/http_agent/http_agent.go:74]: I think maybe just encode it in a POST request.More testing can be done.

**Review by sanxun0325 — CHANGES_REQUESTED** [review]

File: common/http_agent/http_agent.go (hunk 2 above)
Comment (mark4z) [reply common/http_agent/http_agent.go:74]: OK，how about PUT method?

Comment (mark4z) [reply common/http_agent/http_agent.go]: and when I add test method in client_test, the test will failed with no logs, what I should do?

Comment (sanxun0325) [reply common/http_agent/http_agent.go]: dingding:wg46mjc，钉钉聊一下吧

**Pull request closed without merging**